presto-main/src/main/java/com/facebook/presto/sql/planner/plan/AggregationNode.java

```diff
@@ -109,18 +106,38 @@ public AggregationNode(@JsonProperty("id") PlanNodeId id,
         super(id);
 
         this.source = source;
-        this.aggregations = ImmutableMap.copyOf(requireNonNull(aggregations, "aggregations is null"));
-        this.functions = ImmutableMap.copyOf(requireNonNull(functions, "functions is null"));
-        this.masks = ImmutableMap.copyOf(requireNonNull(masks, "masks is null"));
-        for (Symbol mask : masks.keySet()) {
-            checkArgument(aggregations.containsKey(mask), "mask does not match any aggregations");
-        }
+        this.assignments = ImmutableMap.copyOf(requireNonNull(assignments, "aggregations is null"));
         requireNonNull(groupingSets, "groupingSets is null");
         checkArgument(!groupingSets.isEmpty(), "grouping sets list cannot be empty");
         this.groupingSets = ImmutableList.copyOf(groupingSets);
         this.step = step;
         this.hashSymbol = hashSymbol;
         this.groupIdSymbol = requireNonNull(groupIdSymbol);
+
+        ImmutableList.Builder<Symbol> outputs = ImmutableList.builder();
+        outputs.addAll(getGroupingKeys());
+        hashSymbol.ifPresent(outputs::add);
+        outputs.addAll(assignments.keySet());
+
+        this.outputs = outputs.build();
+    }
+
+    /**
+     * @deprecated pass Assignments object instead
+     */
+    @Deprecated
```

Comment:
It would be nice to remove this `@Deprecated` method being used. At least in places where it is trivial. Without this effort those method won't disappear shortly causing this code being messed for longer time.